### ISSUE TITLE
electCandidate always elect the wrong node

### ISSUE THREAD

**Issue by whatot** [mariadb-corporation/replication-manager#43]
The problem is here, the ``max`` is always 0, so the function always eject the latest one.
https://github.com/mariadb-corporation/replication-manager/blob/master/monitor.go#L211

**Comment by tanji** [comment]
I'm sorry; how is it possible that Max is always 0? Max should be equal to the sequence number of the node, and as far as I know, it works well for all people using it.

**Comment by whatot** [comment]
I add debug log here, it always 0

**Comment by tanji** [comment]
I can't see the debug log, sorry.

**Comment by whatot** [comment]
I add extra debug log in the eject step, after line 211.
```
		if verbose {
			logprintf("DEBUG: Election rig: %s [%d] is current gtid is %d, current max is %d", sl.URL, i, seqList[i], max)
		}
```

**Comment by tanji** [comment]
Can you post the output of MRM log while you're doing your failover or switchover?
Lack of info isn't exactly helpful at this time.

**Comment by whatot** [comment]
```
2016-06-23 19:32:56 INFO : Slave db3.nutscloud.com [1] has been elected as a new master                                                                                                                           
 2016-06-23 19:32:56 DEBUG: Election rig: db3.nutscloud.com [1] is current gtid is 270607, current max is 0                                                                                                        
 2016-06-23 19:32:56 DEBUG: Election rig: db1.nutscloud.com [0] is current gtid is 270410, current max is 0                                                                                                        
 2016-06-23 19:32:56 DEBUG: Processing 2 candidates                                                                                                                                                                
 2016-06-23 19:32:55 INFO : Electing a new master                                                                                                                                                                  
 2016-06-23 19:32:55 INFO : Checking long running updates on master                                                                                                                                                
 2016-06-23 19:32:55 INFO : Flushing tables on db2.nutscloud.com (master)                                                                                                                                          
 2016-06-23 19:32:55 INFO : Starting master switch 
```

**Comment by tanji** [comment]
@whatot: Please run MRM with --verbose and --logfile option, and attach the whole log here.
Also please indicate which command you are using to start MRM.

**Comment by whatot** [comment]
`replication-manager --hosts=db1.nutscloud.com,db2.nutscloud.com,db3.nutscloud.com --user=root:rootpass --rpluser=replicator:slavepass --gtidcheck=false --interactive=false --failover=monitor --verbose --logfile="./mrm.log"`
[mrm.txt](https://github.com/mariadb-corporation/replication-manager/files/329842/mrm.txt)

I use C-s to switchover master when sysbench is runing oltp test on the master sever.

I add debug log after the line `var max uint64` line 211.
`
if verbose {
    logprintf("DEBUG: Election rig: %s [%d] is current gtid is %d, current max is %d", sl.URL, i, seqList[i], max)
}
`

The result is that the electCandidate always chooses the latest slave.

**Comment by whatot** [comment]
@tanji  Every `Starting master switch` mean once C-s switchover.

**Comment by tanji** [comment]
OK, sorry, this was fixed in 0.7 but the fix wasn't backported.
I did a 0.6 backport, please use latest release. Fixed in 7000b0a
